thumbnail-service/src/main/java/no/sikt/nva/handler/ThumbnailRequestHandler.java

```diff
@@ -0,0 +1,141 @@
+package no.sikt.nva.handler;
+
+import static nva.commons.core.attempt.Try.attempt;
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestHandler;
+import com.amazonaws.services.lambda.runtime.events.S3Event;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import no.sikt.nva.thumbnail.ThumbnailerException;
+import no.sikt.nva.thumbnail.ThumbnailerManager;
+import no.unit.nva.s3.S3Driver;
+import nva.commons.core.Environment;
+import nva.commons.core.JacocoGenerated;
+import nva.commons.core.StringUtils;
+import nva.commons.core.attempt.Failure;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.core.ResponseInputStream;
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.GetObjectRequest;
+import software.amazon.awssdk.services.s3.model.GetObjectResponse;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+
+public class ThumbnailRequestHandler implements RequestHandler<S3Event, Void> {
+
+    public static final String COULD_NOT_CREATE_THUMBNAIL_LOG_MESSAGE = "Could not create thumbnail";
+    public static final String CURRENTLY_SUPPORTING_THE_FOLLOWING_MIME_TYPES_LOG_MESSAGE = "Currently Supporting the "
+                                                                                           + "following Mime Types: ";
+    public static final String FILENAME_PREFIX = "filename=\"";
+    public static final String FILENAME_POSTFIX = "\"";
+    private static final Logger logger = LoggerFactory.getLogger(ThumbnailRequestHandler.class);
+    private static final String INPUT_FILE_NAME_PREFIX = "/tmp/"; // protects against overwriting existing files in
+    // directory, plus it's the only place aws allows filewriting;
+    private static final String OUTPUT_FILE_NAME = "thumbnail.png";
+    private static final String THUMBNAIL_BUCKET_ENVIRONMENT_FIELD = "THUMBNAIL_BUCKET";
+    public static final String DELIMITER = ", ";
+    private final String thumbnailBucketName;
+    private final S3Client s3Client;
+    private File inputFile;
```

Comment:
Is there a reason for storing inputFile and mimeTypeFromS3Response as an instance field on the handler class? Would it not be better to pass then between the methods? I find it a lot easier to follow and read the code then.